### PR TITLE
Node-side metrics for asynchronous backing

### PR DESCRIPTION
**Pull request by mrcnski**
Closes https://github.com/paritytech/polkadot/issues/6236

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2258028

File: node/network/collator-protocol/src/validator_side/metrics.rs [hunk 1]
```diff
@@ -75,7 +85,7 @@ impl metrics::Metrics for Metrics {
 			collation_requests: prometheus::register(
 				prometheus::CounterVec::new(
 					prometheus::Opts::new(
-						"polkadot_parachain_collation_requests_total",
+						"polkadot_parachain_collation_requests",
```
[the remark this answers is not in the document]
Comment (sandreim) [reply node/network/collator-protocol/src/validator_side/metrics.rs:88]: There has to be a good reason to change this. Such changes can break existing dashboards/monitoring.

File: node/network/collator-protocol/src/validator_side/mod.rs [hunk 2]
```diff
@@ -979,6 +979,8 @@ where
 	Sender: CollatorProtocolSenderTrait,
 	I: IntoIterator<Item = ((ParaId, Hash), Vec<BlockedAdvertisement>)>,
 {
+	state.metrics.time_request_unblocked_collations();
```
Comment (sandreim) [inline node/network/collator-protocol/src/validator_side/mod.rs:982]: Does this actually observe anything? I believe it's dropped immediately.

`let _timer = state.metrics.time_request_unblocked_collations();`

File: node/network/collator-protocol/src/validator_side/metrics.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrcnski) [reply node/network/collator-protocol/src/validator_side/metrics.rs:88]: I thought that might be the case. The inconsistency triggered my OCD, but happy to revert it.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2287087

**Review by rphmeier — APPROVED** [review]

**Pull request merged**